draft-add-requirements.md

```diff
@@ -104,92 +108,137 @@ the IETF may publish, such as DNS-over-QUIC {{?I-D.ietf-dprive-dnsoquic}}.
+
+Clients may also seek to validate the identity of the designated resolver, beyond what is required by the relevant
+protocol. In general, clients may impose arbitrary additional requirements and heuristics as they see fit.
+
+
+## Network-identified designated resolvers
 
 DNS servers are often provisioned by a network as part of DHCP options {{?RFC2132}},
 IPv6 Router Advertisement (RA) options {{?RFC8106}}, Point-to-Point Protocol (PPP) {{?RFC1877}}, or
 3GPP Protocol Configuration Options (TS24.008). Historically this is usually one or more Do53 resolver IP addresses,
 to be used for traditional unencrypted DNS.
 
 A solution is required that enhances the set of information delivered to include details of one or more
-equivalent encrypted DNS resolvers, or states that there are none.
+designated encrypted DNS resolvers, or states that there are none. Such resolvers could be on the local network, somewhere
+upstream, or on the public Internet.
 
-# Resolver-identified encrypted resolvers
+## Resolver-identified designated resolvers
 
 To support cases where the network is unable to identify an encrypted resolver, it should be possible to learn
-the details of one or more equivalent encrypted DNS resolvers by communicating with the network-recommended
+the details of one or more designated encrypted DNS resolvers by communicating with the network-recommended
 unencrypted Do53 resolver. This should involve an exchange that uses standard DNS messages that can be
 handled, or forwarded, by existing deployed software.
 
-It is frequently the case that Do53 resolvers announced by home networks are difficult to upgrade to support encrypted operation.
-In such cases it is possible that the only option for encrypted operation is to refer to a separate globally-addressed
-encrypted DNS resolver.
+Each resolver in the set may be at a different network location, which leads to several subcases for the mapping from Do53
+to a particular designated resolver.
+
+### Local to local
 
 If the local resolver has been upgraded to support encrypted DNS, the client may not initially be aware that its local resolver
 supports it. Discovering this may require communication with the local resolver, or an upstream resolver, over Do53.
 Clients that choose to use this local encrypted DNS gain the benefits of encryption while retaining the benefits
 of a local caching resolver with knowledge of the local topology.
 
+Clients will be aware when the designated resolver has the same IP address as the Do53 (after looking up its name if required). They
+can use this information in their decision-making as to the level of trust to place in the designated resolver.
+In some networks it will not be possible to deploy encrypted DNS on the same IP address, e.g. because of the increased
+resource requirements of encrypted DNS. Discovery solutions should work in the presence of a change of address.
+
```

Comment:
The client may discover both Do53 and Encrypted DNS servers are hosted on the same IP address but the designated resolver is actually hosted on a different IP address and Encrypted DNS traffic is forwarded to this IP address (attacker uses the Do53 IP address as a proxy). 
The assertion provided in DEER is a better approach for authenticated discovery (IP address of Do53 server in the  server certificate. However, it only works for public DNS servers).

Reply:
I think this text is well-aligned with DEER.  This text says "solutions should work in the presence of a change of address", and DEER uses IP certificates to be able to work securely in that case.

Reply:
DEER uses IP address in the certificate to assert both the Do53 and Encrypted DNS server are operated by the same entity. I don't see a similar explanation in this paragraph.

Reply:
I see.  This paragraph is part of the "local to local" section, so the "IP certificate" approach does not apply.  DEER currently does not meet this requirement.

I think we should clarify the text to express that this "change of address" is to a different local address.  I believe it is possible to meet this requirement securely (and I've proposed a change to DEER to enable it), so I support including it, but it may be a topic of debate.

Reply:
The text now says "Discovery solutions should work in the presence of a change to a different local IP address." Is that enough or do we need more discussion on this point?

Reply:
looks good to me.